services/blockstorage/adapter/filesystem/file_system_persistence.go

```diff
@@ -365,12 +365,17 @@ func (f *BlockPersistence) GetResultsBlock(height primitives.BlockHeight) (*prot
 }
 
 func (f *BlockPersistence) getBlockAtHeight(height primitives.BlockHeight) (*protocol.BlockPairContainer, error) {
-	var bpc *protocol.BlockPairContainer
-	err := f.ScanBlocks(height, 1, func(h primitives.BlockHeight, page []*protocol.BlockPairContainer) (wantsMore bool) {
-		bpc = page[0]
-		return false
-	})
-	return bpc, err
+	file, err := os.Open(f.blockFileName())
```

Comment:
this method is identical to GetBlock() you don't need both of them. rename this to `GetBlock` and remove the empty wrapper